compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1159,16 +1159,21 @@ public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
                         new Name(compUnits.get(index++).name)));
                 entry = this.env.scope.lookup(alias);
             }
+            if (!(entry.symbol != null && entry.symbol instanceof BPackageSymbol)) {
```

Comment:
I'm not sure if this is the correct fix. The original error is `class org.wso2.ballerinalang.compiler.semantics.model.symbols.BTypeSymbol cannot be cast to class org.wso2.ballerinalang.compiler.semantics.model.symbols.BPackageSymbol` right? What's `alias` when this happens? What's the type symbol?

I don't think `entry.symbol` can be null here (or below) either. If it is null, it could be due to a bug elsewhere.

Reply:
`boolean` is a example alias for this class cast exception.

Reply:
![boolean_debug](https://user-images.githubusercontent.com/47134066/172543470-856be5d9-ffb3-4fa0-9f42-6604567d82a7.png)

Reply:
transaction alias havent the class cast exception issue

Reply:
![transaction_debug](https://user-images.githubusercontent.com/47134066/172543594-88488116-c9e1-4a20-be28-97502787d05b.png)

Reply:
In the `boolean` alias, `entry.symbol` is a `BTypeSymbol`.
In the `transaction` alias, `entry.symbol` is a `BPackageSymbol`.
`decimal, float, map, string` aliases behave similar to boolean.

Reply:
I guess the issue is that we are getting the builtin types here rather than the imports/modules. 

@chiranSachintha, thoughts please? Did we consider this when introducing predeclared imports?

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/36459#discussion_r893161109